_posts/2019-12-12-live-view-live-component.md

```diff
@@ -0,0 +1,397 @@
+---
+author: Sophie DeBenedetto
+author_link: https://github.com/sophiedebenedetto
+categories: general
+tags: ['live view']
+date: 2019-10-20
+layout: post
+title: LiveView Design Patterns: LiveComponent and the Single Responsibility Principle
+excerpt: >
+  It's easy to end up with an overly complex LiveView that houses lots of business rules and responsibilities. We can use `Phoenix.LiveComponent` to build a LiveView feature that is clean, maintainable and adherent to the Single Responsibility Principle.
+---
+
+## LiveView Can Get Messy  
+
+As LiveView becomes a more established technology, we naturally find ourselves leveraging to back more and more complex features. If we're not careful, this can lead to "fat controller syndrome"––live views that are jam packed with complex business logic and disparate responsibilities, just like the classic "fat Rails controller".
+
+How can we write live views that are easy to reason about and maintain and adhere to common design principles like SRP?
+
+One way to achieve this goal is to leverage the `Phoenix.LiveComponent` behaviour.
+
+## Introducing `Phoenix.LiveComponent`
+
+Components are modules that use the `Phoenix.LiveComponent` behaviour. This behaviour provides
+
+> ...a mechanism to compartmentalize state, markup, and events in LiveView. [*](https://hexdocs.pm/phoenix_live_view/Phoenix.LiveComponent.html)
```

Comment:
An asterisk will probably difficult to click on, more so on a cell phone. You should change it to something with more text :)